### DISCUSSION TITLE
Spoolman database on my own hosting

### DISCUSSION THREAD

**Discussion by jankorous** [Donkie/Spoolman#499] [Ideas]
I found in settings an option to declare my own database where i can store data from Spoolman. Unfortunately, I couldn't find a comprehensive readme anywhere on how and what to set. I had to install Mainsail from scratch due to a printer glitch and lost all data from the local spoolman instance.

How and from where to back up the local database?

What are the options?

**Comment by jeffalanmcclain** [comment]
The default DB is setup as an SQLite DB in the /data directory of your spoolman docker install (check the yaml settings for this directory to customize).  By default, a file is created there the first time you run the service, or you can delete it and swap your own backup from an old system in there.  I've successfully pulled this file over and edited it raw with an SQLite viewer to create aligned ID's to my own DB I had manually created (spool id's and such).

**Comment by Donkie** [comment]
Also see this wiki page: https://github.com/Donkie/Spoolman/wiki/Backup-and-Restore
